### ISSUE TITLE
Migration to free form fortran

### ISSUE THREAD

**Issue by ramirezfranciscof** [nanolebrero/lio#51]
Is there a reason why we are still using fixed form code @nanolebrero ? Because if not, I think it would be a good idea to use free form instead. I mean, we are not using fortran 77 anymore, and I can't say with certainty, but I wouldn't discard that some existing warnings may have to do with bad compatibility.

**Comment by nanolebrero** [comment]
Contesto en castellano (sorry).
El codigo está en formato fijo por razones históricas, usar un formato
libre esta bien, pero cambiarlo no creo que sea tan trivial y es muy
posible que haya que tocar muuuuchas lineas de codigo (en particular se me
ocurren dos casos, cuando se continua una línea en la siguiente con > y
cuando se comenta con C en el primer espacio).
LAs nuevas rutinas pueden estar en el formato libre sin problemas y si
alguien se ofrece a cambiar el formato a todas no me opondré.

Saludos

Nano

2015-08-20 11:59 GMT-03:00 ramirezfranciscof <notifications@github.com>:

**Comment by ramirezfranciscof** [comment]
(Te jode que te siga hablando en inglés? Yo lo hago màs que nada para que no quede chad excluido)

Ok, what I meant by migrating to free format is not to actually rewrite all code, but basically move all extensions from .f to .f90 (and maybe there are some older structures that we will have to change because of this). From what I understand, compilers use different rules to deal with .f and .f90 (.f uses old fortran 77 standars). Again, I do not have a deep knowledge about this, but recently every place I've been reading about fortran makes an emphasis in diferentiating between the still maintained f77 standar and the modern fortran 95/2003/2008.

**Comment by nanolebrero** [comment]
No me jode que hables en inlges, pero a mi me cuesta un poco escribir así
que lo hago en castellano (sorry again Chad).
Hasta donde sé, la diferencia es sólo en el formato. Se pueden invocar
comandos de fortran90 aún en los .f (como allocate y otros). Si le cambias
el nombre de los .f por .F90 el compilador querrá leerlos con otro formato
y habrá un monton de problemas (o tal vez no sean taaaantos). Digamos que
no hay que reescribir todo pero si corregir bastantes líneas, ignoro si no
hay algún traductor automágico.


2015-08-20 15:19 GMT-03:00 ramirezfranciscof <notifications@github.com>:

**Comment by ramirezfranciscof** [comment]
I would think that is the other way around: mixing new features with old fortran 77 should be more error prone when telling the compiler that it is a .f file than when is free format (I mean, probably it will complain about archaic uses that we may need to change, but I am talking more about the subtle, I can't see you untill all has gone to hell kind of bugs). Again, I do not know very much about this...I can try to find out more before doing any modification.

**Comment by emocskos** [comment]
My two cents: What is the gain in this upgrade? I totally agree in
switching to free format for new code, but usually the rule that governs
the rest is "if it's not broke, do not fix it".

I feel it may be a waste of time and a source of bugs (some of them
subtle enough to avoid early detection).

I can't trust in an automatic transduction tool.

Esteban

**Comment by chopkins906** [comment]
(Por favor, hablen en castellano si a ustedes les gustaría.  No hay problemas aquí: puedo leer suficiente para entender el "gist".  Sin embargo, voy a seguir contestar en inglés, porque mi español es mal :)

I tend to agree with @emocskos . It seems that if new routines can be done in free form, I don't see a huge benefit in converting the old code apart from aesthetics / cleaning up warnings.  Personally, I've had some difficulty attempting this type of up-converting old Fortran code before, it didn't end up being as straightforward as I thought it would (though I'm by no means an expert).

**Comment by emocskos** [comment]
Chad spoke, I have a point!! YEAH!!!!!

**Comment by ramirezfranciscof** [comment]
Ok, fair enough. Part of my motivation was that we had some inconveniences when trying to compile both kind of codes (.f and .f90) together with nico, but yesterday I talked with him and apparently it finally worked. Also, I failed to find actual problems that may arise from using new tools in fixed form, so it is true that there may be no reason to do this. Anyways, apologies for the noise and thanks for the feedback!
